tests/runtime/model_management/test_model_manager.py

```diff
@@ -773,6 +774,164 @@ def test_lazy_load_ephemeral_model():
             assert model_id in manager.loaded_models
 
 
+def test_deploy_undeploy_model(deploy_good_model_files):
+    """Test that a model can be deployed by copying to the local models dir"""
+    with TemporaryDirectory() as cache_dir:
+        with non_singleton_model_managers(
+            1,
+            {
+                "runtime": {
+                    "local_models_dir": cache_dir,
+                    "lazy_load_local_models": True,
+                    "lazy_load_poll_period_seconds": 0,
+                },
+            },
+            "merge",
+        ) as managers:
+            manager = managers[0]
+            model_name = "my-model"
+
+            # Make sure model is not currently loaded
+            with pytest.raises(CaikitRuntimeException) as excinfo:
+                manager.retrieve_model(model_name)
+                assert excinfo.value.status_code == grpc.StatusCode.NOT_FOUND
```

Comment:
🤔  I thought that `excinfo` had to be accessed after the scope of the context manager because raising the exception will exit from it. But if that's not the case, then TIL!

Reply:
Now that you say it, I think you're right. That feels wrong to me since it's a contextmanager, but 🤷